sumologic/resource_sumologic_monitors_library_monitor.go

```diff
@@ -90,30 +91,36 @@ func resourceSumologicMonitorsLibraryMonitor() *schema.Resource {
 						"trigger_type": {
 							Type:     schema.TypeString,
 							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{"Critical","Warning","MissingData","ResolvedCritical","ResolvedWarning","ResolvedMissingData"}, false),
 						},
 						"threshold": {
 							Type:     schema.TypeFloat,
 							Optional: true,
 						},
 						"threshold_type": {
-							Type:     schema.TypeString,
-							Optional: true,
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"LessThan", "LessThanOrEqual", "GreaterThan", "GreaterThanOrEqual"}, false),
 						},
 						"time_range": {
 							Type:     schema.TypeString,
 							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{"5m","10m","15m","30m","1h","3h","6h","12h","24h"}, false),
```

Comment:
Also, this is currently not enforced by the Swagger API, only by the backend. We should perhaps add it to the Swagger API as well to maintain consistency?

Reply:
Verified that each of the following is a valid time-range:
`60m`, `1h`, `1d`, `24h`, `23h60m`.

Reply:
The right way to validate it would be to first normalize it by converting to a time range object. For now, I have just added some of the more common synonyms,`60m` and `1d`, to the list.